### PR TITLE
[l10n] Let the welcome letter's language default to admin user

### PR DESCRIPTION
**Pull request by Nienzu**
Currently, the weclome mail for new user is written in default new user language.
We should let the letter written in admin's language.

### CODE REVIEW

File: apps/settings/lib/Mailer/NewUserMailHelper.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ public function __construct(Defaults $themingDefaults,
 	 */
 	public function generateTemplate(IUser $user, $generatePasswordResetToken = false) {
 		$userId = $user->getUID();
-		$lang = $this->l10nFactory->getUserLanguage($user);
+		$lang = $this->l10nFactory->getUserLanguage(\OC::$server->getUserSession()->getUser());
```
Comment (nickvergessen) [inline apps/settings/lib/Mailer/NewUserMailHelper.php:102]: That's not right it.

**Comment by ChristophWurst** [comment]
> Currently, the weclome mail for new user is written in default new user language.

Hmm am I missing something or why isn't the default language the preferred one for new users? Just because the admin localizes their user interface in a certain language doesn't mean the users speak the same.

We have to take out any assumptions. I'd say the default language makes more sense than the admin's language.

**Comment by Nienzu** [comment]
@ChristophWurst 
Currently, the default language for new user is English. In my case, I am using the nextcloud in Chinese Traditional(Taiwan), my users are not familiar with English. So, they will receive the welcome letter in English.
So I suggest the welcome letter would be better written in admin's language. 
But your assumption is reasonable.
Or, could we add a setting in the common setting for admin to choose the language for welcome letter?

**Comment by nickvergessen** [comment]
See https://docs.nextcloud.com/server/21/admin_manual/configuration_server/language_configuration.html

**Pull request closed without merging**